runtime/array.cu

```diff
@@ -20,6 +20,14 @@ struct alignas(sizeof(scalar_t) * align_size) Array {
   __device__ scalar_t& operator[](const unsigned int i) {
     return array[i];
   }
+
+  Array& operator=(scalar_t a[size]) {
+#pragma unroll
+    for (int i = 0; i < size; ++i) {
+      array[i] = a[i];
+    }
+    return *this;
+  }
```

Comment:
Errr... This is so evil, because it can be matched with so many things. For example:
```C++
array = {}
```
will now use this operator, and will IMA because it is trying to dereference a nullptr. Marking as draft again.

Reply:
Fixed. Ready for review again.